### PR TITLE
convert test_virtwho upgrade cases to pytest

### PR DESCRIPTION
**Pull request by hkx303**
**Description**
convert test_virtwho upgrade cases to pytest

**Test Result**
```
(3.8_env) [hkx303@kuhuang robottelo]$ pytest -m "pre_upgrade" tests/upgrades/test_virtwho.py
2021-04-15 20:54:36 - robottelo - INFO - Started setUpClass: robottelo.test/TestCase
==================================== test session starts =====================================
platform linux -- Python 3.8.7, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/hkx303/Documents/CI/robottelo/robottelo, configfile: pyproject.toml
plugins: services-2.2.1, reportportal-5.0.7, xdist-2.2.0, cov-2.11.1, ibutsu-1.13, mock-3.5.1, forked-1.3.0
collecting ... 2021-04-15 20:54:36 - robottelo - DEBUG - Selected 2 and deselected 0 tests based on new infra markers.
2021-04-15 20:54:36 - robottelo - DEBUG - Calling Bugzilla API for {'1802395'}
collected 2 items / 1 deselected / 1 selected  
================== 1 passed, 1 deselected, 15 warnings in 153.18s (0:02:33) ==================


(3.8_env) [hkx303@kuhuang robottelo]$ pytest -m "post_upgrade" tests/upgrades/test_virtwho.py
2021-04-15 21:07:09 - robottelo - INFO - Started setUpClass: robottelo.test/TestCase
==================================== test session starts =====================================
platform linux -- Python 3.8.7, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/hkx303/Documents/CI/robottelo/robottelo, configfile: pyproject.toml
plugins: services-2.2.1, reportportal-5.0.7, xdist-2.2.0, cov-2.11.1, ibutsu-1.13, mock-3.5.1, forked-1.3.0
collecting ... 2021-04-15 21:07:09 - robottelo - DEBUG - Selected 2 and deselected 0 tests based on new infra markers.
2021-04-15 21:07:09 - robottelo - DEBUG - Calling Bugzilla API for {'1802395'}
collected 2 items / 1 deselected / 1 selected       
======================= 1 passed, 1 deselected, 11 warnings in 24.39s ========================
```

### CODE REVIEW

**Review by mshriver — COMMENTED** [review]
I realize you're still working on and testing it, just one small comment about the fixture, otherwise this looks good so far!

File: tests/upgrades/test_virtwho.py [hunk 1]
```diff
@@ -38,7 +38,30 @@
...
+        'hypervisor_username': virtwho.esx.hypervisor_username,
+        'hypervisor_password': virtwho.esx.hypervisor_password,
+        'name': 'preupgrade_virt_who',
+    }
+    return form
+
+
+@pytest.fixture()
+def org_data():
+    data = {'name': 'virtwho_upgrade_org_name'}
```
[the remark this answers is not in the document]
Comment (JacobCallahan) [reply tests/upgrades/test_virtwho.py:60]: i'd say the same is true for the form data fixture

File: tests/upgrades/test_virtwho.py [hunk 2]
```diff
@@ -171,19 +166,17 @@ def test_post_crud_virt_who_configuration(self):
             2. the config and guest connection have the same status.
             3. virt-who config should update and delete successfully.
         """
-        org = entities.Organization().search(query={'search': f'name={self.org_name}'})[0]
+        org = entities.Organization().search(query={'search': f"name={org_data['name']}"})[0]
 
         # Post upgrade, Verify virt-who exists and has same status.
         vhd = entities.VirtWhoConfig(organization_id=org.id).search(
-            query={'search': f'name={self.name}'}
+            query={'search': f"name={form_data['name']}"}
```
Comment (JacobCallahan) [inline tests/upgrades/test_virtwho.py:173]: Please switch these to single-quote strings.

File: tests/upgrades/test_virtwho.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hkx303) [reply tests/upgrades/test_virtwho.py]: Fixed

File: tests/upgrades/test_virtwho.py (hunk 2 above)
Comment (hkx303) [reply tests/upgrades/test_virtwho.py]: Hey, Jake, a little confused, why we have to switch single-quote strings?

Comment (hkx303) [reply tests/upgrades/test_virtwho.py]: Done, thanks

Comment (hkx303) [reply tests/upgrades/test_virtwho.py]: Hey, Jake, a little confused, why we have to switch it to single-quote strings?

Comment (JacobCallahan) [reply tests/upgrades/test_virtwho.py]: Hey there Kunxin, single quotes are the standard defined in robottelo's code standards doc.
https://github.com/SatelliteQE/robottelo/blob/master/docs/code_standards.rst

**Comment by hkx303** [comment]
Added the test result

**Review by JacobCallahan — APPROVED** [review]

File: tests/upgrades/test_virtwho.py (hunk 2 above)
Comment (hkx303) [reply tests/upgrades/test_virtwho.py]: Got it, Thank you

**Review by mshriver — APPROVED** [review]

**Pull request merged**